### PR TITLE
[chore] Log possible duplicate detectors

### PR DESCRIPTION
**Pull request by mcastorina**

### CODE REVIEW

**Comment by mcastorina** [comment]
Example output:
```
2023-04-17T16:41:22-05:00       info-0  trufflehog      loaded decoders {"count": 2}
2023-04-17T16:41:22-05:00       info-0  trufflehog      loaded detectors        {"total": 736, "verification_enabled": 736, "verification_disabled": 0}
2023-04-17T16:41:22-05:00       info-0  trufflehog      possible duplicate detector configured    {"detector": "Square"}
```

This "Square" detector is actually part of our list of detectors [square](https://github.com/trufflesecurity/trufflehog/blob/main/pkg/detectors/square/square.go#L92) and [squareapp](https://github.com/trufflesecurity/trufflehog/blob/main/pkg/detectors/squareapp/squareapp.go#L98), but I'm not sure if it's intentional or not.

File: pkg/engine/engine.go [hunk 1]
```diff
@@ -149,13 +150,28 @@ func Start(ctx context.Context, options ...EngineOption) *Engine {
...
-	ctx.Logger().V(2).Info("loaded detectors",
+	ctx.Logger().Info("loaded decoders", "count", len(e.decoders))
+	ctx.Logger().Info("loaded detectors",
 		"total", len(e.detectors[true])+len(e.detectors[false]),
 		"verification_enabled", len(e.detectors[true]),
 		"verification_disabled", len(e.detectors[false]),
 	)
 
+	// Sanity check detectors for duplicate configuration. Only log in case
+	// a detector has been configure in a way that we isn't represented by
```
Comment (rosecodym) [inline pkg/engine/engine.go:161]: nit: a couple of typos

Comment (rosecodym) [reply pkg/engine/engine.go:161]: also, for my own education, what does "configured in a way that isn't represented by the type and version" mean?

**Review by rosecodym — APPROVED** [review]

File: pkg/engine/engine.go (hunk 1 above)
Comment (mcastorina) [reply pkg/engine/engine.go:161]: Right now we're assuming the identity of a detector is uniquely captured by its type and version, but it's possible for a detector to be configured in a way such that the type and version are the same, but it scans in two different ways.

**Comment by SumoSumir** [comment]
Hey @mcastorina 
Can you update the brew image?
It still has this issue

**Comment by mcastorina** [comment]
@SumoSumir the change in this PR was released in `v3.32.0` on April 20th. According to brew, `trufflehog` is on `v3.33.0` built from source on April 28th.

```
» brew info trufflehog
==> trufflesecurity/trufflehog/trufflehog: stable 3.33.0
Find credentials all over the place
https://github.com/trufflesecurity/trufflehog
/opt/homebrew/Cellar/trufflehog/3.33.0 (5 files, 57.7MB) *
  Built from source on 2023-04-28 at 14:16:20
From: https://github.com/trufflesecurity/homebrew-trufflehog/blob/HEAD/trufflehog.rb
```

Could you verify your trufflehog version and open an issue describing the problem you're seeing? Thanks!

**Pull request merged**